loris/transforms.py

```diff
@@ -303,6 +303,25 @@ def _run_transform(self, target_fp, image_request, image_info, transform_cmd, fi
             crop=False
         )
 
+    def _process(self, cmd, target_fp, image_request, image_info, fifo_fp):
+        process = multiprocessing.Process(
+            target=self._run_transform,
+            kwargs={
+                'target_fp': target_fp,
+                'image_request': image_request,
+                'image_info': image_info,
+                'transform_cmd': cmd,
+                'fifo_fp': fifo_fp
+            }
+        )
+        process.start()
+        process.join(self.transform_timeout)
+        if process.is_alive():
+            logger.info('terminating process for %s, %s',
+                image_info.src_img_fp, target_fp)
+            process.terminate()
+            raise TransformException('JP2 transform process timed out')
```

Comment:
Maybe worth logging how long it timed out after? i.e.

```suggestion
            raise TransformException('JP2 transform process timed out after %d seconds', self.transform_timeout)
```